orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/VtrList.java

```diff
@@ -0,0 +1,159 @@
+package uk.gov.di.orchestration.shared.entity;
+
+import com.google.gson.annotations.JsonAdapter;
+import net.minidev.json.JSONArray;
+import net.minidev.json.parser.JSONParser;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.orchestration.shared.serialization.VtrListAdapter;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.text.MessageFormat.format;
+import static net.minidev.json.parser.JSONParser.DEFAULT_PERMISSIVE_MODE;
+
+@JsonAdapter(VtrListAdapter.class)
+public class VtrList {
+
+    private static final Logger LOG = LogManager.getLogger(VtrList.class);
+
+    public static final VtrList DEFAULT_VTR_LIST = new VtrList(Collections.emptyList());
+
+    private final List<VectorOfTrust> vtr;
+    private final CredentialTrustLevel credentialTrustLevel;
+    private final List<LevelOfConfidence> levelsOfConfidence;
+    private final VectorOfTrust effectiveVectorOfTrust;
+
+    public VtrList(List<VectorOfTrust> vtr) {
+        validateVtr(vtr);
+        this.vtr = !vtr.isEmpty() ? vtr : List.of(VectorOfTrust.DEFAULT_VECTOR_OF_TRUST);
+        this.credentialTrustLevel =
+                this.vtr.stream()
+                        .map(VectorOfTrust::getCredentialTrustLevel)
+                        .min(Comparator.naturalOrder())
+                        .orElseThrow(); // Should never throw.
+        this.levelsOfConfidence =
+                this.vtr.stream()
+                        .filter(vot -> vot.getCredentialTrustLevel() == this.credentialTrustLevel)
+                        .map(VectorOfTrust::getLevelOfConfidence)
+                        .distinct()
+                        .sorted(Comparator.naturalOrder())
+                        .toList();
+        this.effectiveVectorOfTrust =
+                this.vtr.stream()
+                        .filter(
+                                vot ->
+                                        vot.getCredentialTrustLevel()
+                                                .equals(this.credentialTrustLevel))
+                        .filter(vot -> vot.getLevelOfConfidence().equals(levelsOfConfidence.get(0)))
+                        .findFirst()
+                        .orElseThrow(); // Should never throw.
```

Comment:
I added the comments because I'd tend to think orElseThrow suggests perhaps some state or input from the calling code could trigger the exception whereas I'm only using it here because sonar can't tell .findfirst or .min will never return and empty in these cases because the streams always have an element. Is there a better way of doing this? It's not great that sonar effectively forces the use of less clear code.

Reply:
I'll remove them for now.